### ISSUE TITLE
Can't update from WSL to WSL2 on x86-64, error message "The Parameter is Incorrect." (Partial solution)

### ISSUE THREAD

**Issue by nhobson99** [microsoft/WSL#6085]
# Environment

```none
Windows build number: Win32NT 10.0.19041.0 Microsoft Windows NT 10.0.19041.0
Your Distribution version: Ubuntu 20.04 (problem affects all distributions on the Microsoft store)
WSL 2
```

# Steps to reproduce

Disable/Enable Windows Subsystem for Linux in Optional Windows Features.
Disable/Enable Virtual Machine Platform in Optional Windows Features (also tried Hyper-V and Windows Hypervisor Platform, as well as combinations of the three).
Reboot.
Enable SVM option in BIOS (motherboard MSI B350m PRO VDH, latest BIOS version, CPU supports virtualization).
Install the WSL 2 kernel update.
Reboot.
Run the command 'wsl --set-version Ubuntu 2'.

**WSL logs**: 

#  Expected behavior

Running the command "wsl --set-version Ubuntu 2" should convert the current installation of Ubuntu to WSL2.

# Actual behavior

```
PS C:\Users\<username> wsl --set-version Ubuntu 2
Conversion in progress, this may take a few minutes...
For information on key differences with WSL 2 please visit https://aka.ms/wsl2
The parameter is incorrect.
```

# Additional Information
I have made sure I'm using the latest BIOS supported on my motherboard, and that my CPU (AMD Ryzen 5 1500x) supports virtualization. SVM is enabled. I found no issues on GitHub that were identical to mine. Some were similar, but the solutions to those issues did not solve my problem. I have uninstalled and reinstalled the WSL 2 kernel update several times.

From what I can gather, error message "The parameter is incorrect" is often associated with a problem in the Hypervisor/Virtual Machine Platform.

# UPDATE
To see how I fixed my specific problem (network issue, not ARM64-based, so your mileage may vary), see https://github.com/microsoft/WSL/issues/6085#issuecomment-708858336

**Comment by licanhua** [comment]
possible dup of #6031. Can you try create a network manually to see if there is error from Hyper-V

**Comment by nhobson99** [comment]
licanhua, would you mind elaborating? What network am I creating with Hyper-V? I'd be happy to give it a shot.

**Comment by licanhua** [comment]
You can create a [external network](https://www.sysnettechsolutions.com/en/create-external-virtual-switch-in-hyper-v-manager/#:~:text=%20How%20to%20Add%20an%20External%20Virtual%20Switch,card%20of%20your%20physical%20computer%20that...%20More%20), then delete it.
If you didn't see any error, please [collect wsl log, then share the log link](https://github.com/microsoft/WSL/blob/master/CONTRIBUTING.md#8-collect-wsl-logs), and I can help to see where the error is coming from.
If you see an error, it's definitely something wrong with Hyper-V , and you may get better support from [community](https://answers.microsoft.com/en-us/)

**Comment by nhobson99** [comment]
Well, I created the virtual network switch in Hyper-V Manager with all default settings, then I had an idea. I unplugged my USB WIFI adapter that I typically use for my home network, then ran the conversion again, and it worked! Not sure actually if it was the virtual network switch itself that fixed it, or not being on a USB-WIFI connection, but thank you for putting me on the right path.

**Comment by joshkerr** [comment]
I'm also having this error. I'm using a Microsoft Surface Pro X with the new SQ2 chip. There are no options to install Hyper-V. I also don't have the Hyper-V manager. I had no problem upgrading my WSL 1 to 2 distros on my Surface Pro X with the SQ1 chip. Could this issue be related to the new hardware?

**Comment by licanhua** [comment]
I received this comment From: nhobson99, but I don't know why I can't see it in this thread
Sent: Tuesday, October 13, 2020 6:27 PM
Subject: Re: [microsoft/WSL] Can't update from WSL to WSL2, error message "The Parameter is Incorrect." (#6085)

Well, I created the virtual network switch in Hyper-V Manager with all default settings, then I had an idea. I unplugged my USB WIFI adapter that I typically use for my home network, then ran the conversion again, and it worked! Not sure actually if it was the virtual network switch itself that fixed it, or not being on a USB-WIFI connection, but thank you for putting me on the right path.

**Comment by licanhua** [comment]
@joshkerr If you are with Windows Home edition, there is no Hyper-V manger, but you should still see Virtual Machine Platform feature.
Hyper-V manager is not a must have for WSL 2, so it's OK to only have Virtual Machine Platform enabled.
You can [collect wsl log](https://github.com/microsoft/WSL/blob/master/CONTRIBUTING.md#8-collect-wsl-logs), then I can help you to investigate if it's hyper-v network issue or not.

**Comment by tamoanxx** [comment]
I two am having the same issue on my new surface pro x SQ 2

**Comment by joaomoreno** [comment]
Yup this just broke on my Surface Pro X as well. It was working just fine yesterday. I have tried many things including restarting, uninstalling all distros, removing WSL. Nothing seems to fix it.

```
Installing, this may take a few minutes...
WslRegisterDistribution failed with error: 0x80070057
Error: 0x80070057 The parameter is incorrect.

Press any key to continue...
```

**Comment by tamoanxx** [comment]
I am on an Surface Pro X with SQ2 I have uninstall several of the updates that installed and I even tried on the Insider Channels. 

![image](https://user-images.githubusercontent.com/3393558/96018815-4fbdbe00-0e11-11eb-9bcf-c01000898b4d.png)

**Comment by joshkerr** [comment]
I did the same. I’m currently on Insider Beta and still getting the same error.

**Comment by joaomoreno** [comment]
These were the three updates:

![image](https://user-images.githubusercontent.com/22350/96034652-20836d00-0e62-11eb-9c1e-b4a49983791b.png)

**Comment by benhillis** [comment]
We've identified the bug that introduced this regression and will work on servicing this ASAP.

**Comment by tamoanxx** [comment]
@benhillis Do you have any specific way that we can uninstall KB4579311. I am getting this error: 

![image](https://user-images.githubusercontent.com/3393558/96068732-50327500-0e62-11eb-8b95-ba3e9f0a1c91.png)

**Comment by nhobson99** [comment]
I successfully upgraded to WSL2 on both my laptop and my desktop now, and the problem ended up being networking related. The conversion would sometimes partially work when I unplugged my USB WIFI adapter, but then the OS hung when plugging it back in. I tried creating a Hyper-V network, but got a "generic error" when trying to do so (which is why I deleted my original comment saying I'd fixed it, since that was actually misleading information.)

What eventually worked permanently was uninstalling WSL, Hyper-V, Windows Hypervisor and Virtual Machine Platform all from optional windows features, uninstalling network drivers from device manager, rebooting, reinstalling all hypervisor and WSL features, and rebooting again. Finally the changes stuck, and I can use WSL2 on my desktop again. Will update if the change is no longer permanent.

What made this error unique was that there was no actual error number, just a generic-looking error message. The error message itself is from Hyper-V, which really needs a better error message than "The Parameter is Incorrect."

**Comment by licanhua** [comment]
> We've identified the bug that introduced this regression and will work on servicing this ASAP. In the meantime I'd suggest reverting this update (and pausing future updates until this is fixed).

This only applies to KB4579311 + Surface Pro X. "The Parameter is Incorrect." is a different issue if you are not with Surface Pro X and this KB

**Comment by asampal** [comment]
> 
> 
> We've identified the bug that introduced this regression and will work on servicing this ASAP. In the meantime I'd suggest reverting this update (and pausing future updates until this is fixed).

Is the fix specific to addressing the problem on Surface devices or would it also correct the issue I observed in #6091 where I wasn't using a Surface?

**Comment by ninuwa** [comment]
I assume it's specific to ARM64, my Samsung Qualcomm has the same problem after updating.

**Comment by licanhua** [comment]
yes. to be exactly, it's ARM64 related

**Comment by nhobson99** [comment]
I believe another issue should be created for ARM-based systems, since that was not in the initial scope of my issue. The problem should get more attention that way, and should help with taking attention away from the problem I was having initially. licanhua or benhillis, could you change the "bug" and "fixinbound" tags for this issue? As I don't believe the original issue was actually solved, which was about Hyper-V parameters and the vague error message. Thanks.

**Comment by licanhua** [comment]
Agreed, I re-opened #6090 to track the ARM64 issue

**Comment by dennisameling** [comment]
Can confirm that uninstalling KB4579311 through the Control Panel fixes the issue on the Surface Pro X (arm64), have paused updates for now.

**Comment by Alovchin91** [comment]
Same problem here after installing KB4579311 on Surface Pro X (SQ1).

**Comment by ac1982** [comment]
I have the same problem here. pause the update right now ```KB4579311 on Surface Pro X```

**Comment by nhobson99** [comment]
Changed the title of this issue to dissuade people commenting with different issues. If you are on x86 and not ARM, see https://github.com/microsoft/WSL/issues/6085#issuecomment-708858336

**Comment by Alovchin91** [comment]
@nhobson99  Maybe also reopen https://github.com/microsoft/WSL/issues/6090 then?

**Comment by nhobson99** [comment]
More @licanhua or @benhillis, I'm not a dev so I don't actually have the authority to reopen a closed issue. Either way, these issues are used internally to track regressions, but there is a bug fix already inbound for ARM based devices, so I don't believe it needs to be reopened (as it is aptly labeled "fix inbound"). Just trying to get attention for ARM devices into the right place.

**Comment by Alovchin91** [comment]
> More @licanhua or @benhillis, I'm not a dev so I don't actually have the authority to reopen a closed issue. Either way, these issues are used internally to track regressions, but there is a bug fix already inbound for ARM based devices, so I don't believe it needs to be reopened (as it is aptly labeled "fix inbound"). Just trying to get attention for ARM devices into the right place.

Oh, sorry, I thought you were a moderator 😅 The ARM issue is not fixed for the end users (us) yet, so it's strange to me that it's closed.

**Comment by falloutphil** [comment]
Confirmed issue seen on Intel Core i9-9900K @ 3.60GHz - an identical Windows upgrade and import of the same WSL tar was rolled out across several machines - I didn't see an the issues on PCs build from exactly the same version of v2004  (Intel Xenon CPUE E4-1225 v5  @ 3.30 GHz, or Intel Core i7-6700 @ 3.40GHz).  As I only have one i9 I can't conclusively say that this is the factor by testing on a second alas.  It's definitely not (just) an ARM issue tho.

**Comment by mirogl** [comment]
> > More @licanhua or @benhillis, I'm not a dev so I don't actually have the authority to reopen a closed issue. Either way, these issues are used internally to track regressions, but there is a bug fix already inbound for ARM based devices, so I don't believe it needs to be reopened (as it is aptly labeled "fix inbound"). Just trying to get attention for ARM devices into the right place.
> 
> Oh, sorry, I thought you were a moderator 😅 The ARM issue is not fixed for the end users (us) yet, so it's strange to me that it's closed.

Hi @falloutphil @Alovchin91 - it is fixed now for arm64 - see https://github.com/microsoft/WSL/issues/6090#issuecomment-715459214

**Comment by falloutphil** [comment]
Thanks - I'm assuming this ticket will now track the different issue with the same result that we're seeing on intel chipsets?

> 
> 
> Hi @falloutphil @Alovchin91 - it is fixed now for arm64 - see [#6090 (comment)](https://github.com/microsoft/WSL/issues/6090#issuecomment-715459214)

**Comment by mirogl** [comment]
> Thanks - I'm assuming this ticket will now track the different issue with the same result that we're seeing on intel chipsets?
> 
> > Hi @falloutphil @Alovchin91 - it is fixed now for arm64 - see [#6090 (comment)](https://github.com/microsoft/WSL/issues/6090#issuecomment-715459214)

I'm not a moderator or involved with Microsoft - but yes - https://github.com/microsoft/WSL/issues/6090 is arm64+KB4579311 related and this ticket is for x86/x64.

**Comment by ChristianBauerEng** [comment]
> 
> 
> I successfully upgraded to WSL2 on both my laptop and my desktop now, and the problem ended up being networking related. The conversion would sometimes partially work when I unplugged my USB WIFI adapter, but then the OS hung when plugging it back in. I tried creating a Hyper-V network, but got a "generic error" when trying to do so (which is why I deleted my original comment saying I'd fixed it, since that was actually misleading information.)
> 
> What eventually worked permanently was uninstalling WSL, Hyper-V, Windows Hypervisor and Virtual Machine Platform all from optional windows features, uninstalling network drivers from device manager, rebooting, reinstalling all hypervisor and WSL features, and rebooting again. Finally the changes stuck, and I can use WSL2 on my desktop again. Will update if the change is no longer permanent.
> 
> What made this error unique was that there was no actual error number, just a generic-looking error message. The error message itself is from Hyper-V, which really needs a better error message than "The Parameter is Incorrect."

Can confirm this. After disabling the builtin WiFi network adapter in device manager Hyper-V automatically created new network adapters. After this I could upgrade the distro and re-enable the WiFi adapter afterwards.
